tests/spark/test_spark_model_export.py

```diff
@@ -823,7 +824,8 @@ def test_shutil_copytree_without_file_permissions(tmpdir):
 def test_log_model_with_code_paths(spark_model_iris):
     artifact_path = "model"
     with mlflow.start_run(), mock.patch(
-        "mlflow.spark._add_code_from_conf_to_system_path"
+        "mlflow.spark._add_code_from_conf_to_system_path",
+        wraps=_add_code_from_conf_to_system_path,
```

Comment:
Why do we add `wraps`?

Reply:
Because the test case used to pass even though the impl was broken due to the mock. Now the mock is a spy; this test case fails on `master`, as it should, and passes on the PR branch.

Reply:
Makes sense!

Reply:
Btw why are we using mock here? to avoid mutating `sys.path`?

Reply:
Presumably to verify that the code was added to the sys path